pynicotine/config.py

```diff
@@ -699,6 +699,30 @@ def _set_config(self):
                     self.sections[section][option] = use_speed_limit
                     continue
 
+                # Map legacy folder/user grouping modes (3.1.0)
+                if (section, option) in [
+                    ("searches", "group_searches"),
+                    ("transfers", "groupdownloads"),
+                    ("transfers", "groupuploads")
+                ]:
+                    mode = self.sections[section].get(option, "folder_grouping")
+
+                    if mode == "0":
+                        mode = "ungrouped"
+
+                    elif mode == "1":
+                        mode = "folder_grouping"
+
+                    elif mode == "2":
+                        mode = "user_grouping"
+
+                    # Verify mode validity
+                    elif mode not in {"ungrouped", "folder_grouping", "user_grouping"}:
+                        mode = "folder_grouping"
```

Comment:
We can remove this code. We should only take care of migrating old config values.